test/xray.js

```diff
@@ -0,0 +1,183 @@
+'use strict';
+
+const sinon = require('sinon');
+const { expect } = require('chai');
+const AWSXRay = require('aws-xray-sdk');
+
+const XRay = require('../lib/xray');
+
+describe('XRay', function () {
+  beforeEach(function () {
+    this.resolve = sinon.stub().resolves();
+    this.reject = sinon.stub().rejects();
+    this.log = {
+      warn: Function.prototype
+    };
+
+    sinon.stub(this.log);
+
+    this.context = {
+      log: this.log
+    };
+  });
+
+  afterEach(function() {
+    sinon.verifyAndRestore();
```

Comment:
1. There are no mocks in this test suite, so no need to verify.
2. Other tests use `this.sandbox`. Yes, I know that it's not needed in the new `sinon`, but I prefer to keep it consistent and switch from `sandbox` to just `sinon` in a separate PR.